packages/astro/src/core/build/static-build.ts

```diff
@@ -161,7 +161,7 @@ async function ssrBuild(opts: StaticBuildOptions, internals: BuildInternals, inp
 			// SSR needs to be last
 			ssr && vitePluginSSR(internals, settings.adapter!),
 		],
-		envPrefix: 'PUBLIC_',
+		envPrefix: settings.config.envPrefix ?? 'PUBLIC_',
```

Comment:
I don't think we need to add our own top-level `envPrefix` option, we just need to support Vite's `envPrefix` option.

```suggestion
		envPrefix: viteConfig.envPrefix ?? 'PUBLIC_',
```